core/src/main/java/tech/tablesaw/analytic/AggregateFunction.java

```diff
@@ -1,16 +1,45 @@
 package tech.tablesaw.analytic;
 
-public interface AggregateFunction<T, R> {
+/**
+ * Analytic Aggregate function.
+ *
+ * <p>See {@link AggregateFunctions} for more background.
+ *
+ * <p>This class allows callers to independently slide a the left side, and right side of the
+ * window. Implementers must keep track of the state of the window as it slides.
+ *
+ * @param <T> the type of the values in the input column (E.G. Integer).
+ * @param <R> the type of values in the return column (E.G. Double).
+ */
+interface AggregateFunction<T, R> {
 
+  /**
+   * Slides the left side (aka start) of the window window to the right (aka end) by removing the
+   * first element in the window. E.G. [(1, 2, 3, 4, 5)] -> [1, (2, 3, 4, 5)]
+   */
   void removeLeftMost();
 
+  /**
+   * Slides the right side (aka end) of the window to the right (aka end) by appending a new missing
+   * element. E.G. [(1, 2, 3), NULL, 5] -> [(1, 2, 3, NULL), 5)]
+   */
```

Comment:
this comment and the one below seem to be reversed.

Reply:
It is ok. This is adding the NULL element on the right side of the window.

Reply:
Sorry. I still don't understand how a method called:

_addRIghtMost(T value)_ can be usefully described as "... appending a new missing element"

while

_addRightMostMissing()_ can be described as "...appending a new non-missing element"

The method names state the opposite of the comments. Also, if you're always adding NULL in the first method, why does it have a parameter at all? The implication of addRightMost(T value) is that you're adding some value of type T, rather than null.

Reply:
You are absolutely right. I do not know how I missed that.

Reply:
thats fine. glad we got it sorted.